src/callbacks_step/glm_speed.jl

```diff
@@ -0,0 +1,76 @@
+
+"""
+    GlmSpeedCallback(; glm_scale=0.5, cfl=1.0)
+
+Update the divergence cleaning wave speed `c_h` according to the time step
+computed in [`StepsizeCallback`](@ref) for the ideal GLM-MHD equations.
+The `cfl` number should be set to the same value as for the time step size calculation. The
+`glm_scale` ensures that the GLM wave speed is lower than the fastest physical waves in the MHD
+solution and should thus be set to a value within the interval [0,1]. Note that `glm_scale` = 0
+deactivates the divergence cleaning.
+"""
+struct GlmSpeedCallback{RealT<:Real}
+  glm_scale::RealT
+  cfl::RealT
+end
+
+
+function Base.show(io::IO, cb::DiscreteCallback{Condition,Affect!}) where {Condition, Affect!<:GlmSpeedCallback}
+  glm_speed_callback = cb.affect!
+  @unpack glm_scale, cfl = glm_speed_callback
+  print(io, "GlmSpeedCallback(glm_scale=", glm_scale, ", cfl=", cfl, ")")
+end
+
+
+function Base.show(io::IO, ::MIME"text/plain", cb::DiscreteCallback{Condition,Affect!}) where {Condition, Affect!<:GlmSpeedCallback}
+  if get(io, :compact, false)
+    show(io, cb)
+  else
+    glm_speed_callback = cb.affect!
+
+    setup = [
+             "GLM wave speed scaling" => glm_speed_callback.glm_scale,
+             "Expected CFL number" => glm_speed_callback.cfl,
+            ]
+    summary_box(io, "GlmSpeedCallback", setup)
+  end
+end
+
+
+function GlmSpeedCallback(; glm_scale=0.5, cfl=1.0)
+  # when is the callback activated
+  condition = (u, t, integrator) -> true
+
+  glm_speed_callback = GlmSpeedCallback(glm_scale, cfl)
+
+  @assert(0<=glm_speed_callback.glm_scale<=1,"glm_scale must be between 0 and 1")
```

Comment:
```suggestion
  @assert 0 <= glm_speed_callback.glm_scale <=1 "glm_scale must be between 0 and 1"

  glm_speed_callback = GlmSpeedCallback(glm_scale, cfl)
```
I think we should check validty before creating any new data structures.

Reply:
works for me. Also you do not need parentheses around the assert...good to know